src/Core/src/Platform/Android/ButtonExtensions.cs

```diff
@@ -10,6 +11,35 @@ namespace Microsoft.Maui
 {
 	public static class ButtonExtensions
 	{
+		public static void UpdateStrokeColor(this MaterialButton nativeButton, IButtonStroke buttonStroke)
+		{
+			var stroke = buttonStroke.StrokeColor?.ToNative();
+
+			if (stroke is not null)
+			{
+				var states = new int[][] {
+					new int[] { Android.Resource.Attribute.StateEnabled },
+					new int[] {-Android.Resource.Attribute.StateEnabled },
+					new int[] {-Android.Resource.Attribute.StateChecked },
+					new int[] { Android.Resource.Attribute.StatePressed }
+				};
+				var c = (int)stroke;
+				nativeButton.StrokeColor = new ColorStateList(states, new int[] { c, c, c, c });
+			}
+		}
+
+		public static void UpdateStrokeThickness(this MaterialButton nativeButton, IButtonStroke buttonStroke)
+		{
+			if (buttonStroke.StrokeThickness >= 0)
+				nativeButton.StrokeWidth = (int)(buttonStroke.StrokeThickness * (nativeButton?.Resources?.DisplayMetrics?.Density ?? 1));
```

Comment:
What do we want to do when we get -1? Just a no-op, or should it collapse to a 0 just so that it does almost what is expected. I would think that setting the thickness to say 10, and then -1 would remove the border?

Reply:
Should this also be using the `ToPixels` extension method?